sw/host/opentitanlib/src/test_utils/load_sram_program.rs

```diff
@@ -329,32 +334,53 @@ pub fn load_sram_program(jtag: &Rc<dyn Jtag>, file: &SramProgramFile) -> Result<
 /// execution from the program buffer (PMP entry 13),
 /// subsequent instruction fetches will generate exceptions.
 ///
-/// Debug module concepts like abstract commands and the program buffer
-/// buffer are defined in "RISC-V External Debug Support Version 0.13.2"
-/// [2]. OpenTitan's (vendored-in) implementation lives in
-/// hw/vendor/pulp_riscv_dbg.
+/// Debug module concepts like abstract commands and the program buffer are
+/// defined in "RISC-V External Debug Support Version 0.13.2" [2]. OpenTitan's
+/// (vendored-in) implementation lives in hw/vendor/pulp_riscv_dbg.
 ///
 /// [0]: https://opentitan.org/book/sw/device/silicon_creator/rom/doc/memory_protection.html
 /// [1]: https://github.com/lowRISC/opentitan/issues/14978
 /// [2]: https://riscv.org/wp-content/uploads/2019/03/riscv-debug-release.pdf
 /// [3]: https://github.com/lowRISC/opentitan/blob/master/hw/ip/rv_core_ibex/rtl/ibex_pmp_reset.svh
-pub fn prepare_epmp_for_sram(jtag: &Rc<dyn Jtag>) -> Result<()> {
-    log::info!("Configure ePMP for SRAM execution");
+pub fn prepare_epmp(jtag: &Rc<dyn Jtag>) -> Result<()> {
+    // Setup ePMP for SRAM execution.
+    log::info!("Configure ePMP for SRAM execution.");
     let pmpcfg3 = jtag.read_riscv_reg(&RiscvReg::Csr(RiscvCsr::PMPCFG3))?;
     log::info!("Old value of pmpcfg3: {:x}", pmpcfg3);
-    // write "L NAPOT X W R" to pmp3cfg in pmpcfg3
+    // Write "L NAPOT X W R" to pmpcfg3 in region 15.
     let pmpcfg3 = (pmpcfg3 & 0x00ffffffu32) | 0x9f000000;
     log::info!("New value of pmpcfg3: {:x}", pmpcfg3);
     jtag.write_riscv_reg(&RiscvReg::Csr(RiscvCsr::PMPCFG3), pmpcfg3)?;
-    // write pmpaddr15 to map the SRAM range
+    // Write pmpaddr15 to map the SRAM range.
     // hex((0x10000000 >> 2) | ((0x20000 - 1) >> 3)) = 0x4003fff
     let base = top_earlgrey::SRAM_CTRL_MAIN_RAM_BASE_ADDR as u32;
     let size = top_earlgrey::SRAM_CTRL_MAIN_RAM_SIZE_BYTES as u32;
-    // make sure that this is a power of two
+    // Make sure that this is a power of two.
     assert!(size & (size - 1) == 0);
     let pmpaddr15 = (base >> 2) | ((size - 1) >> 3);
     log::info!("New value of pmpaddr15: {:x}", pmpaddr15);
     jtag.write_riscv_reg(&RiscvReg::Csr(RiscvCsr::PMPADDR15), pmpaddr15)?;
+
+    // Setup ePMP for R/W access to MMIO region.
+    log::info!("Configure ePMP for MMIO access.");
+    let pmpcfg2 = jtag.read_riscv_reg(&RiscvReg::Csr(RiscvCsr::PMPCFG2))?;
+    log::info!("Old value of pmpcfg2: {:x}", pmpcfg2);
+    // Write "L TOR X W R" to pmpcfg2 in region 11.
+    let pmpcfg2 = (pmpcfg2 & 0x00ffffffu32) | 0x8f000000;
+    log::info!("New value of pmpcfg2: {:x}", pmpcfg2);
+    jtag.write_riscv_reg(&RiscvReg::Csr(RiscvCsr::PMPCFG2), pmpcfg2)?;
+    // Write pmpaddr10 and pmpaddr11 to map the MMIO range.
+    let base = top_earlgrey::MMIO_BASE_ADDR as u32;
+    let size = top_earlgrey::MMIO_SIZE_BYTES as u32;
+    // make sure that this is a power of two
+    assert!(size & (size - 1) == 0);
```

Comment:
If we're sure the MMIO region is power-of-two sized, why are we using a TOR region instead of a NAPOT region?

Reply:
This was done to match the configuration in the ROM: https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/rom_epmp_init.S;drc=fbd4be8b952dd9fbd29e2e3398ef749e599608b1;l=98